src/goto-programs/string_refine_preprocess.cpp

```diff
@@ -0,0 +1,1343 @@
+/*******************************************************************\
+
+Module: Preprocess a goto-programs so that calls to the java String
+        library are recognized by the string solver
+
+Author: Romain Brenguier
+
+Date:   September 2016
+
+\*******************************************************************/
+
+#include "string_refine_preprocess.h"
+
+#include <util/std_expr.h>
+#include <util/symbol_table.h>
+#include <util/pointer_offset_size.h>
+#include <util/prefix.h>
+#include <goto-programs/class_identifier.h>
+#include <solvers/refinement/string_expr.h>
```

Comment:
@peterschrammel do you prefer to have these changes (on string_expr and refined_string_type) in this pull request or should it be a new one?